ivy/functional/frontends/numpy/statistics/order_statistics.py

```diff
@@ -35,6 +39,15 @@ def _cpercentile(N, percent, key=lambda x: x):
     return d0 + d1
 
 
+@to_ivy_arrays_and_back
+@handle_numpy_out
+def ptp(a, axis=None, out=None, dtype=None, keepdims=False):
+    x = ivy.max(a, axis=axis, keepdims=keepdims)
+    y = ivy.min(a, axis=axis, keepdims=keepdims)
+    ret = ivy.subtract(x, y)
+    return ret.astype(dtype, copy=False)

```

Comment:
Will cast the returned value to input dtype 
```suggestion
    return ret.astype(a.dtype, copy=False)

```